src/executor/plan_executor.cpp

```diff
@@ -105,38 +52,110 @@ void PlanExecutor::ExecutePlan(
+                       LOG_TRACE("column content: [%s]", str.c_str());
+                       result.push_back(std::move(str));
+                     }
+                   }
+                   on_complete(execute_result, std::move(result));
+                 });
+}
+
+static void InterpreteAndExecutePlan(
+    std::shared_ptr<planner::AbstractPlan> plan,
+    concurrency::TransactionContext *txn,
+    const std::vector<type::Value> &params,
+    const std::vector<int> &result_format,
+    std::function<void(executor::ExecuteResult, std::vector<ResultValue> &&)>
+        on_complete) {
+  executor::ExecuteResult p_status;
+  std::vector<ResultValue> result;
+
+  std::unique_ptr<executor::ExecutorContext> executor_context(
+      new executor::ExecutorContext(txn, params));
+
+  bool status;
+  std::unique_ptr<executor::AbstractExecutor> executor_tree(
+      BuildExecutorTree(nullptr, plan.get(), executor_context.get()));
+
+  status = executor_tree->Init();
+  if (status != true) {
+    p_status.m_result = ResultType::FAILURE;
+    CleanExecutorTree(executor_tree.get());
+    on_complete(std::move(p_status), std::move(result));
+    return;
+  }
+
+  // Execute the tree until we get result tiles from root node
+  while (status == true) {
+    status = executor_tree->Execute();
+    std::unique_ptr<executor::LogicalTile> tile(executor_tree->GetOutput());
+
+    // Some executors don't return logical tiles (e.g., Update).
+    if (tile.get() != nullptr) {
+      LOG_TRACE("Final Answer: %s", tile->GetInfo().c_str());
+      std::vector<std::vector<std::string>> tuples;
+      tuples = tile->GetAllValuesAsStrings(result_format, false);
+
+      // Construct the returned results
+      for (auto &tuple : tuples) {
+        for (unsigned int i = 0; i < tile->GetColumnCount(); i++) {
+          LOG_TRACE("column content: %s", tuple[i].c_str() != nullptr
+                                              ? tuple[i].c_str()
+                                              : "-emptry-");
+          result.push_back(std::move(tuple[i]));
+        }
+      }
     }
   }
   p_status.m_processed = executor_context->num_processed;
   p_status.m_result = ResultType::SUCCESS;
-  p_status.m_result_slots = nullptr;
-  return;
+  CleanExecutorTree(executor_tree.get());
+
+  on_complete(std::move(p_status), std::move(result));
+}
+
+void PlanExecutor::ExecutePlan(
+    std::shared_ptr<planner::AbstractPlan> plan,
+    concurrency::TransactionContext *txn,
+    const std::vector<type::Value> &params,
+    const std::vector<int> &result_format,
+    std::function<void(executor::ExecuteResult, std::vector<ResultValue> &&)>
+        on_complete) {
+  PL_ASSERT(plan != nullptr && txn != nullptr);
+  LOG_TRACE("PlanExecutor Start (Txn ID=%" PRId64 ")", txn->GetTransactionId());
+
+  if (!settings::SettingsManager::GetBool(settings::SettingId::codegen) ||
+      !codegen::QueryCompiler::IsSupported(*plan)) {
```

Comment:
It's my fault for writing it this way, but can you change the condition to something less convoluted?
```c++
bool codegen_enabled = settings::SettingsManager::GetBool(settings::SettingId::codegen);
if (codegen_enabled && codegen::QueryCompiler::IsSupported(*plan)) {
  // Compile
} else {
  // Interpret
}
```